# Um pouco de estatística

In [18]:
import pandas as pd

url = 'http://personal.tcu.edu/kylewalker/data/colleges.csv'
data = pd.read_csv(url, encoding = 'latin_1')
data.shape

(7804, 122)

Observamos que nosso conjunto de dados possui mais de 7800 linhas e 122 colunas! Neste ponto, embora você ainda não saiba quais insights estão contidos nos dados, já deve ter uma ideia das perguntas de pesquisa que lhe interessam. Como essas perguntas provavelmente não exigem todas as 122 colunas, é seguro restringir o conjunto de dados apenas às colunas necessárias para sua análise. Nossa principal pergunta de pesquisa é:

**Como a proporção de estudantes alunos de graduação com 25 anos ou mais varia entre faculdades e universidades que possuem bacharelado?**

Para responder a essa pergunta, precisamos identificar algumas colunas essenciais e filtrar os dados de acordo. As colunas que manteremos são as seguintes:

- **INSTNM**: Nome da instituição;
- **STABBR**: Estado onde a instituição está localizada;
- **PREDDEG**: Tipo principal de diploma concedido pela instituição (códigos: 1 para certificados, 2 para diplomas de associado, 3 para bacharelados e 4 para pós-graduação);
- **CONTROL**: Propriedade da instituição (códigos: 1 para pública sem fins lucrativos, 2 para privada sem fins lucrativos, e 3 para privada com fins lucrativos);
- **UGDS**: Número de alunos de graduação matriculados na instituição;
- **UG25abv**: Percentual de alunos de graduação com 25 anos ou mais na instituição.


In [19]:
cols = ['INSTNM', 'STABBR', 'PREDDEG', 'CONTROL', 'UGDS', 'UG25abv']
df = data.filter(cols)

df.columns = [col.lower() for col in df.columns]

df.rename(columns={'stabbr': 'state', 'instnm': 'institution'}, inplace=True)

dfNotNull = df.dropna()

In [20]:
dfNotNull['ugds'].mean()

2351.1247333238516

In [21]:
dfNotNull['ugds'].median()

441.0

In [51]:
df1 = pd.DataFrame(dfNotNull.groupby(['control','preddeg'])['ugds'].median(),index=None)
df2 = pd.DataFrame(dfNotNull.groupby(['control','preddeg'])['ugds'].mean(),index=None)

In [56]:
df1['média'] = df2['ugds']

In [63]:
df1.replace('ugds','mediana')

ugds        média
control preddeg                     
1       0          44.5   126.700000
        1         503.0  2207.334432
        2        3930.5  6317.664103
        3        6867.0  9769.071429
2       0          26.5   375.833333
        1         118.0   210.743961
        2         426.0   829.879518
        3        1357.5  2120.670230
        4          18.0   104.222222
3       0         152.0   186.560606
        1         128.0   254.363636
        2         369.0   662.347009
        3         562.0  2455.136519
        4          48.0    48.000000

In [70]:
df.query('ugds > 5000')

institution state  preddeg  control     ugds  \
1     University of Alabama at Birmingham    AL        3        1  11200.0   
3     University of Alabama in Huntsville    AL        3        1   5525.0   
4                Alabama State University    AL        3        1   5354.0   
5               The University of Alabama    AL        3        1  28692.0   
9                       Auburn University    AL        3        1  19761.0   
...                                   ...   ...      ...      ...      ...   
7760          DeVry University-California    CA        3        3   7436.0   
7764            DeVry University-Illinois    IL        3        3  20112.0   
7784          University of North Georgia    GA        3        1  14502.0   
7799    Central Georgia Technical College    GA        1        1   7146.0   
7802     Arizona State University-Skysong    AZ        3        1   8227.0   

      ug25abv  
1      0.2422  
3      0.2640  
4      0.1270  
5      0.0853  
9      0.0415  
...       ...  
7760   0.6393  
7764   0.7671  
7784   0.1623  
7799   0.5555  
7802   0.6616  

[899 rows x 6 columns]

In [69]:
df[df['ugds'] == 33]

institution state  preddeg  \
630       Universal College of Beauty Inc-Los Angeles 1    CA        1   
1112                     Concept College of Cosmetology    IL        1   
1196  Educators of Beauty College of Cosmetology-La ...    IL        1   
2082  Cosmetology Careers Unlimited College of Hair ...    MN        1   
2363                     La'James International College    NE        1   
2732  Mesivta of Eastern Parkway-Yeshiva Zichron Mei...    NY        0   
3178                  Northern Institute of Cosmetology    OH        1   
4124               Evans Hairstyling College-Cedar City    UT        1   
4550                    Rogies School of Beauty Culture    PR        1   
5107                                 Bristol University    CA        1   
5780                             My Le's Beauty College    CA        1   
5876     National Graduate School of Quality Management    MA        3   
5883    Orleans Niagara BOCES-Practical Nursing Program    NY        1   
6166                            Xtreme Career Institute    PR        1   
6185               Culpeper Cosmetology Training Center    VA        1   
6406                        Daymar College-Madisonville    KY        2   
6428                               Ideal Beauty Academy    IN        1   
6512                         InterCoast Colleges-Carson    CA        1   
6692                International Polytechnic Institute    CA        1   
6719                   University of Aesthetics-Chicago    IL        1   
6781                            Central Nursing College    CA        1   
6905                      University of Fort Lauderdale    FL        2   
6915                     Master Educators Beauty School    ID        1   
6930                       Avalon School of Cosmetology    MN        1   
6959                        Aiken School of Cosmetology    SC        1   
7248               The Salon Professional Academy-Elgin    IL        1   
7289                                Mesivta Keser Torah    NJ        0   

      control  ugds  ug25abv  
630         3  33.0   0.4242  
1112        3  33.0   0.3333  
1196        3  33.0   0.1818  
2082        3  33.0   0.3939  
2363        3  33.0   0.2424  
2732        2  33.0      NaN  
3178        3  33.0   0.3939  
4124        3  33.0   0.2424  
4550        3  33.0   0.2424  
5107        3  33.0   0.4242  
5780        3  33.0   0.7576  
5876        2  33.0   1.0000  
5883        1  33.0   0.8485  
6166        3  33.0   0.6970  
6185        1  33.0   0.2424  
6406        3  33.0   0.6970  
6428        3  33.0   0.5455  
6512        3  33.0   0.4242  
6692        2  33.0   0.5455  
6719        3  33.0   0.7576  
6781        3  33.0   0.6364  
6905        2  33.0   0.9697  
6915        3  33.0   0.1515  
6930        3  33.0   0.4545  
6959        3  33.0   0.5758  
7248        3  33.0   0.4848  
7289        2  33.0      NaN